me.hyliu.fluentreader.yml

```diff
@@ -0,0 +1,91 @@
+app-id: me.hyliu.fluentreader
+runtime: org.freedesktop.Platform
+runtime-version: '20.08'
+branch: stable
+sdk: org.freedesktop.Sdk
+base: org.electronjs.Electron2.BaseApp
+base-version: '20.08'
+sdk-extensions:
+  - org.freedesktop.Sdk.Extension.node14
+command: start-fluent-reader
+separate-locales: false
+finish-args:
+  - --share=ipc
+  - --socket=x11 # TODO: add wayland when this merged: https://github.com/electron/electron/issues/10915
+  - --device=dri # hardware acceleration
+  - --socket=pulseaudio # videos/audios
+  - --share=network
+  - --filesystem=host # TODO: remove after xdg-desktop-portal merged: https://github.com/electron/electron/pull/19159
+  - --talk-name=org.freedesktop.Notifications # TODO: remove after xdg-desktop-portal merged
+modules:
+  - name: fluent-reader
+    buildsystem: simple
+    build-options:
+      # Add the node bin directory.
+      append-path: /usr/lib/sdk/node14/bin
+      env:
+        # Set the cache directory. No need for ELECTRON_CACHE
+        XDG_CACHE_HOME: /run/build/fluent-reader/flatpak-node/cache
+    build-commands:
+      # Use the offline mirror.
+      - 'npm install --offline --prefix=fluent-reader
+        --cache=$FLATPAK_BUILDER_BUILDDIR/flatpak-node/npm-cache'
+      # Download the packages.
+      # If you were using npm with electron-webpack/electron-builder, then the above two commands
+      # would look more like the npm commands in the vanilla-quick-start manifest, just without
+      # the --prefix.
+
+      # Apply the build command patch
+      - git apply package.json.patch --directory fluent-reader
+
+      # Run electron-builder, passing the architecture arguments to it.
+      # Note that the -- is important; without that, the argument will be passed to
+      # yarn instead of electron-builder.
+      - >
+        . flatpak-node/electron-builder-arch-args.sh; npm run
+        --prefix=fluent-reader --offline build; npm run --prefix=fluent-reader
+        --offline package-linux-unpacked -- $ELECTRON_BUILDER_ARCH_ARGS
+      # Copy the resulting, unpacked directory to /app.
+      # (A glob is used because the directory name may contain the current arch.)
+      - 'cp -r fluent-reader/bin/linux/*/linux*unpacked /app/fluent-reader'
+      # If you passed --electron-non-patented-ffmpeg, you could install it like this:
+      # - 'install -Dm 755 flatpak-node/libffmpeg.so -t /app/electron-webpack-quick-start'
+      # Install the wrapper script to start it.
+      - 'install -Dm 755 start-fluent-reader.sh /app/bin/start-fluent-reader'
+      # Installing SVG Icon is sufficient
+      - install -Dm644 fluent-reader/dist/icons/logo.svg
+        /app/share/icons/hicolor/scalable/apps/me.hyliu.fluentreader.svg
+    sources:
+      - type: git
+        url: https://github.com/yang991178/fluent-reader
+        commit: master
```

Comment:
This should use specific commit or tag.

Reply:
Does that mean the flatpak metadata has to be updated by hand every time there are changes? I observed some apps are updated by flathubbot. Is there any way to do so?

Reply:
You should package a released version, therefor use one of the tags, or a branch + specific commit. So yes.

For a given revision of this flathub manifest repository you should always get the same exact result.

Reply:
Done